File: 045_Triangular_Penatagonal_Hexagonal/Solution.go
package main

import (
	"fmt"
	"math"
)

func isTriangular(x int) bool {
	return math.Sqrt(float64(1+8*x)) == math.Floor(math.Sqrt(float64(1+8*x)))
}

func isPentagonal(x int) bool {
	return math.Sqrt(float64(1+24*x)) == math.Floor(math.Sqrt(float64(1+24*x))) && int(math.Sqrt(float64(1+24*x))+1)%6 == 0
}

func main() {
	var n, k, b int
	fmt.Scan(&n, &k, &b)

	var penMax, hexMax int
	if k == 3 {
		penMax = int((math.Sqrt(float64(1+24*n)) + 1) / 6)
		for i := 1; i < penMax; i++ {
			p := i * (3*i - 1) / 2
			if isTriangular(p) {
				fmt.Println(p)
			}
		}
	} else {
		hexMax = int((math.Sqrt(float64(1+8*n)) + 1) / 4)
		for i := 1; i < hexMax; i++ {
			h := i * (2*i - 1)
			if isPentagonal(h) {
				fmt.Println(h)
			}
		}
	}
}

File: 045_Triangular_Penatagonal_Hexagonal/Solution.py
def is_triangular(x):
    return True if (1+8*x)**0.5==int((1+8*x)**0.5) else False
def is_pentagonal(x):
    return True if (1+24*x)**0.5==int((1+24*x)**0.5) and int((1+24*x)**0.5+1)%6==0 else False

n,k,b = map(int, input().strip().split())

pen_max=int((1+24*n)**0.5+1)//6
hex_max=int((1+8*n)**0.5+1)//4
if k==3:
    for i in range(1,pen_max):
        p=i*(3*i-1)//2
        if is_triangular(p):
            print(p)
else:
    for i in range(1,hex_max):
        h=i*(2*i-1)
        if is_pentagonal(h):
            print(h)
        

